fex2/collect.py

```diff
@@ -1,5 +1,77 @@
+from __future__ import annotations
+
 import csv
 import re
+from typing import Generator, Callable, Dict, List
+
+CONST_HEADER_KEYWORD = '[FEX2_HEADER] '
+CONST_EXPERIMENT_KEYWORD = '[FEX2_EXPERIMENT] '
+
+
+class Experiments:
+
+    def __init__(self, experiments_output: str, default_parsers=True):
+        self.__experiments_output = experiments_output
+        self.parsers: Dict[str, Callable[[Experiment], None]] = {}
+        if default_parsers:
+            self.parsers.update({
+                'benchmark': lambda experiment: get_run_argument('benchmark', experiment.header),
+                'thread_count': lambda experiment: get_run_argument('thread_count', experiment.header),
+                'type': lambda experiment: get_run_argument('type', experiment.header).split('_')[0],
+                'subtype': lambda experiment: "_".join(get_run_argument('type', experiment.header).split('_')[1:])})
+        # main header
+        self.type: str
+        self.name: str
+        with open(self.__experiments_output, 'r') as infile:
+            for line in infile:
+                if line.startswith(CONST_HEADER_KEYWORD):
+                    self.type = get_run_argument('experiment_type', line)
+                    self.name = get_run_argument('name', line)
+                    break
```

Comment:
Add an `else` statement that would raise an exception if the header is missing.

Reply:
Good point :+1: